### DISCUSSION TITLE
should I update to php-vips v2

### DISCUSSION THREAD

**Discussion by goldmerc** [libvips/php-vips#188] [Q&A]
Hi, just wondering whether it's sensible to update from v1 to v2?

How long can we expect v1 to be supported?

Are there any particular benefits to using v2?

Thanks,

Jonathan

**Comment by goldmerc** [comment]
Just following up, it seems the main benefit of sticking to v1 would be to wait for v2 to support FFI preloading. Is there any news on this? Thanks, Jonathan

**Comment by jcupitt** [comment]
v1 used to cause a lot of install problems for users, and was quite a lot of effort to maintain.

v2 ought to be much simpler:

- there's no binary module to build
- you don't need all the libvips dev headers, you only need the shared library
- you don't need a C compiler
- it works on Windows
- it's easier to develop, since the whole thing is written in php
- it supports every version of libvips with no recompiling 

We don't plan to kill v1, but equally we're not planning any further development of it. 

Support for preload would be nice, but I don't have time to work on it at the moment. If someone else would like to have a go, a PR would be very welcome.

**Comment by goldmerc** [comment]
Thanks @jcupitt - that's clear

Sorry to say that I don't have the skillset to sort out the preload. I'm just a fairly poor hobby coder ;-)
